### PR TITLE
Use upstream containerd 1.2.4

### PR DESCRIPTION
**Pull request by randomvariable**
Signed-off-by: Naadir Jeewa <jeewan@vmware.com>

### CODE REVIEW

File: build/amis/ansible/roles/containerd/defaults/main.yml [hunk 1]
```diff
@@ -13,4 +13,6 @@
 # limitations under the License.
 ---
 containerd_version: "1.2.4"
-containerd_sha256: 329d44850685f8b511779c00763df50dd24257b3a1c594aff57e77bcee1b1269
+containerd_sha256: "8a2dc3cd9b2226f9703072ab0c18caca9a965d7e3bbf0de20e333406be513e78"
+cri_tools_version: "1.13.0"
```
Comment (detiber) [inline build/amis/ansible/roles/containerd/defaults/main.yml:17]: I'm wondering if we should leave these unset here and set/override them through packer like we do with `kubernetes_version` and `kubernetes_cni_version`, that is something that I can do as a followup, though.

**Comment by randomvariable** [comment]
I'm done here.

File: build/amis/ansible/roles/containerd/defaults/main.yml (hunk 1 above)
Comment (randomvariable) [reply build/amis/ansible/roles/containerd/defaults/main.yml:17]: Very possibly. Agree it should be done as follow up though.

File: build/amis/ansible/roles/containerd/tasks/main.yml [hunk 2]
```diff
@@ -32,18 +32,58 @@
...
+    mode: 0755
+    owner: root
+    group: root
+    dest: /usr/local/bin
+    extra_opts:
+      - --no-overwrite-dir
+
+- name: install systemd unit for containerd
+  copy:
+    src: "{{ role_path }}/files/etc/systemd/system/containerd.service"
```
Comment (detiber) [inline build/amis/ansible/roles/containerd/tasks/main.yml:60]: Does `etc/systemd/system/containerd.service` not work here?

Comment (detiber) [reply build/amis/ansible/roles/containerd/tasks/main.yml:60]: Actually, there shouldn't be a need to nest the `containerd.service` file so deeply under `files`, could just put it directly under `files`

File: build/amis/ansible/roles/containerd/files/etc/crictl.yaml [hunk 3]
```diff
@@ -0,0 +1 @@
+runtime-endpoint: unix:///run/containerd/containerd.sock
```
Comment (detiber) [inline build/amis/ansible/roles/containerd/files/etc/crictl.yaml:1]: We should probably configure containerd to use systemd as the cgroup manager. I can do that as a followup.

File: build/amis/ansible/roles/containerd/tasks/main.yml (hunk 2 above)
Comment (detiber) [reply build/amis/ansible/roles/containerd/tasks/main.yml:60]: I can handle that as a followup.

Comment (randomvariable) [reply build/amis/ansible/roles/containerd/tasks/main.yml:60]: Happy for you to just change it before finalising the upstream PR. This is just to your PR.

**Pull request merged**